app/components/elements/Dialog.js

```diff
@@ -24,13 +24,21 @@ export function DialogTitle(props) {
     ...dialogTitleProps
   } = props;
 
+  const ICON_WIDTH = 28;
+
   return (
     <Flex
       p={3}
-      sx={{ alignItems: 'center', justifyContent: 'space-between', borderBottom: props.divider ? borders.divider : 'unset', ...sx }}
+      sx={{ alignItems: 'center', borderBottom: props.divider ? borders.divider : 'unset', ...sx }}
       {...dialogTitleProps}
     >
-      {children}
+      <Flex
+        // offset by Icon width to ensure Title is properly centered when closeIcon is present
+        pl={closeIcon ? ICON_WIDTH : 0}
```

Comment:
This is somewhat a hack to offset the width of the icon when trying to center the title
![image](https://github.com/user-attachments/assets/6a10c0ea-57ce-4319-be6d-2ba33a47d967)


I couldn't think of a better way to easily do it. Typically I would set the icon to `position: absolute`, but I was unable to make that happen with the `sx` prop.

Reply:
I think that, as far as CSS hacks go, this is pretty readable for future readers, which is about the best you can hope for. 😄

Reply:
I missed this similar PR to the one for WEB-2802, where I left the following comment:

| It's tricky, but I believe the best way to do have middle piece of content be true-centered in-between elements of potentially variable sizes with flex is to use flex-grow: 1, flex-basis: 0 on the outer elements.

| I ended up having to do this on the Dialog title to accommodate adding a back button to some modals as part of the data connection work. [Here's a link to the commit](https://github.com/tidepool-org/blip/pull/1485/commits/15bc53b554512f24df2d833541c67df34b2d67bb).

I recommend the flex-based approach, as it will be necessary when we add a potential back button and text on the other side of the title.

Reply:
Thanks @clintonium-119, that seems to work!

![image](https://github.com/user-attachments/assets/1edf26b4-84db-4f70-9394-dfa7a0cd163c)

If you don't mind, I've cherry-picked your commit into my branch c7603f2977439e099e0b33c9adf43711c8456575

Reply:
Thanks @clintonium-119 and @krystophv for the review :)

The approach works great. If you don't mind, I've cherry-picked your commit into my branch to avoid merge error, since your commit only touches this file. 

c7603f2977439e099e0b33c9adf43711c8456575

![image](https://github.com/user-attachments/assets/1edf26b4-84db-4f70-9394-dfa7a0cd163c)